include/aspect/material_model/rheology/compositional_viscosity_prefactors.h

```diff
@@ -0,0 +1,108 @@
+  {
+    using namespace dealii;
+
+    namespace Rheology
+    {
+      /**
+       * A class that handles multiplication of viscosity for a given compositional
+       * field. The multiplication factors for each composition (viscosity
+       * prefactors) are also declared, parsed, and in some cases calculated in this class.
+       */
+      template <int dim>
+      class CompositionalViscosityPrefactors : public ::aspect::SimulatorAccess<dim>
+      {
+        public:
+          /**
+           * Constructor.
+           */
+          CompositionalViscosityPrefactors();
+
+          /**
+           * Declare the parameters this function takes through input files.
+           */
+          static
+          void
+          declare_parameters (ParameterHandler &prm);
+
+          /**
+           * Read the parameters from the parameter file.
+           */
+          void
+          parse_parameters (ParameterHandler &prm);
+
+          /**
+           * Compute the viscosity.
+           */
+          double
+          compute_viscosity (const MaterialModel::MaterialModelInputs<dim> &in,
+                             const double base_viscosity,
+                             const unsigned int composition_index,
+                             const unsigned int q) const;
+
+        private:
+          /**
+           * The viscosity prefactors or terms used to calculate the viscosity
+           * prefactors, which are read in from the input file by the
+           * parse_parameters() function. Users can choose between different schemes.
+           * none: no viscosity change
+           * hk04_olivine_hydration: calculate the viscosity change due to hydrogen
+           * incorporation into olvine using Hirth & Kohlstaedt 2004 10.1029/138GM06.
+           * This method requires a composition called 'bound_fluid' which is required
+           * by the reactive fluid transport material model to weaken the viscosity with
+           * elevated ratios of H/Si ppm in the solid.
+           * The prefactor for a given compositional field is multiplied with a
+           * base_viscosity value provided by the material model, which
+           * is then returned to the material model.
+           * Constant values
+           */
+          enum ViscosityPrefactorScheme
+          {
+            none,
+            hk04_olivine_hydration
+          } viscosity_prefactor_scheme;
+
+          // Initialize variables for the water fugacity calculation, from HK04
+          std::vector<double> water_fugacity_exponents;
+          // From Hirth & Kohlstaedt 2004, equation 6
+          const double activation_energy_H2O = 40e3; // J/mol/K
+          const double activation_volume_H2O = 10e-6; // m^3/mol
+
+          // We calculate the Molar mass of olivine using the molar mass of fayalite (203.79) and the
+          // molar mass of forsterite (140.693), and a mass fraction of 90% forsterite.
+          const double M_olivine = 0.14727; // kg/mol
```

Comment:
```suggestion
          const double mass_olivine = 0.14727; // kg/mol
```

In line with other renamed variables above, perhaps go from `M` to `mass`